src/app/portlets/dot-edit-page/content/dot-edit-content.component.ts

```diff
@@ -73,50 +72,18 @@ export class DotEditContentComponent implements OnInit, OnDestroy {
     ngOnInit() {
         this.dotLoadingIndicatorService.show();
 
-        this.dotMessageService
-            .getMessages([
-                'editpage.content.contentlet.remove.confirmation_message.message',
-                'editpage.content.contentlet.remove.confirmation_message.header',
-                'editpage.content.contentlet.add.content',
-                'dot.common.message.saving',
-                'dot.common.message.saved'
-            ])
-            .subscribe();
-
-        this.dotEditContentHtmlService.iframeActions.subscribe((contentletEvent: any) => {
-            this.ngZone.run(() => {
-                this.iframeActionsHandler(contentletEvent.name)(contentletEvent);
-            });
-        });
-
-        this.dotEditContentHtmlService.pageModelChange.filter((model) => model.length).subscribe((model) => {
-            if (this.originalValue) {
-                this.ngZone.run(() => {
-                    this.isModelUpdated = !_.isEqual(model, this.originalValue);
-                });
-            } else {
-                this.setOriginalValue(model);
-            }
-        });
-
-        this.route.parent.parent.data.pluck('content').subscribe((pageState: DotRenderedPageState) => {
-            this.setPageState(pageState);
-        });
-
-        this.swithSiteSub = this.switchSiteSubscription();
-        this.loadEditModePageSub = this.loadEditPageEventSubcription();
-
+        this.getMessages();
+        this.setInitalData();
         this.setDialogSize();
+        this.subscribeSwitchSite();
+        this.subscribeIframeCustomEvents();
+        this.subscribeIframeActions();
+        this.subscribePageModelChange();
     }
 
     ngOnDestroy(): void {
-        if (this.loadEditModePageSub) {
-            this.loadEditModePageSub.unsubscribe();
-        }
-
-        if (this.swithSiteSub) {
-            this.swithSiteSub.unsubscribe();
-        }
+        this.destroy$.next(true);
```

Comment:
This is a method to "auto unsubscribe" from all subscribe at once, I saw it in a video in the ngconf and here is a post with how it works: https://alligator.io/angular/takeuntil-rxjs-unsubscribe/